TOOLS/ce/src/main/scala/org.opalj.ce/ConfigList.scala

```diff
@@ -0,0 +1,77 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package ce
+
+import scala.collection.mutable.ListBuffer
+
+/**
+ * Stores a List structure inside the ConfigNode structure
+ * @param entries contains a List of ConfigNodes
+ * @param comment are all the comments associated with the List
+ */
+case class ConfigList(entries: ListBuffer[ConfigNode], comment: Comment) extends ConfigNode {
+    /**
+     * Formats the entry into HTML code
+     * @param label required if the List is part of an object (Writes the key of the K,V Map there instead). Overrides the label property of the Comment object.
+     * @param HTMLHeadline accepts the HTML syntax of the Headline of the value. Can contain $ label and $ brief flags for filling with content.
+     * @param HTMLContent accepts the HTML syntax of the content frame for the value. Must contains a $ content flag for correct rendering.
+     * @return returns the Config List as HTML code
+     */
+    override def toHTML(label: String, HTMLHeadline: String, HTMLContent: String, sorted: Boolean): String = {
+        var HTMLString = ""
+        var head = label
+        if (this.comment.label.nonEmpty) head = this.comment.label
+
+        // Get HTML data for all child Nodes
+        var content = "<p>" + comment.toHTML + "</p>"
+        for (entry <- entries) {
+            content += entry.toHTML("", HTMLHeadline, HTMLContent, sorted)
+        }
+
+        // Adds Header line with collapse + expand options
+        HTMLString += HTMLHeadline.replace("$label", head).replace("$brief", this.comment.getBrief)
+
+        // Add content below
+        HTMLString += HTMLContent.replace("$content", content)
+
+        HTMLString
+    }
+
+    /**
+     * Checks if the list is empty
+     * @return true if both the List and the comment are empty
+     */
+    override def isEmpty: Boolean = {
+        for (entry <- entries) {
```

Comment:
Use `entries.exists` (or `entries.forall`)

Reply:
This is not sufficient, because the entries may exist too, but in am empty state. It needs to be recursed.
e.g.
`key = [
     [],
     ""
]`
In this case, an empty List exists inside of another list, together with a value. While the entries.exists would return true, the objects hold exactly 0 information and this, "key" should return false. "forall" is a nice thing that I did not know yet

Reply:
No, `exists` takes a lambda function, it is the logical exists quantifier. You code is 100% semantically the same as
`if(!entries.forall(_.isEmpty)) return false`

In fact, I'd rewrite the whole method as
```
override def isEmpty: Boolean =
    comment.isEmpty && entries.forall(_.isEmpty)
```

Reply:
Implemented